src/cirrus/github_tools.py

```diff
@@ -269,7 +269,25 @@ def tag_release(self, tag, master='master', push=True):
             raise RuntimeError(msg)
         self.repo.create_tag(tag)
         if push:
-            self.repo.remotes.origin.push(self.repo.head, tags=True)
+            count = 0
+            error_flag = None
+            while count < attempts:
+                error_flag = None
+                try:
+                    self.repo.remotes.origin.push(self.repo.head, tags=True)
+                    break
+                except Exception as ex:
+                    msg = "Error pushing tags: {}".format(ex)
+                    LOGGER.info(msg)
+                    error_flag = ex
+                    time.sleep(cooloff)
```

Comment:
nah, its really just to get a small window to pass some network weather or a server being a dick. Most of the examples I have seen of this failing re-run almost as fast as I can restart the build.